### ISSUE TITLE
[Feature Request]: Please Add backdrop-filter support to be used with transparent window (so to recreate fluent design acrylic material)

### ISSUE THREAD

**Issue by alixsep** [electron/electron#30412]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

I tried using 

```css
body {
    background-color: #0000002f;
    -webkit-backdrop-filter: blur(8px);
}
```
And the `background-color` worked but the `-webkit-backdrop-filter` didn't.

### Proposed Solution

being able to blur the window background by `-webkit-backdrop-filter`

### Alternatives Considered

[This package](https://github.com/arkenthera/electron-vibrancy/blob/master/README.md) seem to be helpful but it is deprecated and there is no linux support ...

### Additional Information

I am sure that with the fluent design trend, more devs may want to follow this design pattern, so this feature will be absolutely necessary.

**Comment by sadpixel** [comment]
### _tldr;_
**When Microsoft finalizes Windows 11 & its new WinUI APIs, Electron will be able to tap into these and automagically allow your BrowserWindow to request that the OS make it acrylic or mica.**

I'd estimate this is easily another year out, irrespective of when Windows 11 goes RTM. I do not expect these APIs to become stable and public for non-first-party & win32 apps for a while.



### _The longer version:_

What you propose is NOT possible given a variety of technical reasons – especially in a consistent, reliable, predictable & multi-platform manner. The CSS "backdrop-filter" property is and forever will be limited to the domain of Chromium's renderer. It can only perform graphical transformations like that on its own content. It does not, cannot and will never have access to whatever is "underneath" the Electron window. Please forget about *this* as the solution to what you're trying to accomplish!

That said, no need to despair! There will *eventually* be a solution, you just need to be patient.
It is helpful to understand the following:

1. Acrylic (and, on Windows 11, its sibling Mica) do much, much more than just blur the background. These modes are very context-dependent and are fine-tuned for a variety of edge cases to minimize the chance of the effect leading to transformations that impair legibility or become unacceptably distracting. (They are also optimized for performance beyond anything the Chromium renderer could ever be hoped to reasonably provide!) The same is true for the various Vibrancy modes available on macOS (which Electron has, for a while now, supported naively).

2. There is a less-than-ideal "solution" to this in the form of a native node module/DLL that you have to compile into your project from the source. It does forbidden, dark magic with undocumented, private Microsoft APIs to request that a window's chrome be acrylic-ized. This code predates Windows 11 so it may only work on Windows 10, and I don't know how it handles rounded window corners. If you look around on GitHub and Google, you'll probably find it, but it's not worth your time, IMO, for a shipping project. (With all due respect to the coder who creatively hacked together this workaround!)

3. If you need this *today* and can't wait, I'd suggest rolling your own WinUI/XAML/.NET/WPF app with Microsoft's new Edge WebView2 control. You'll lose all the benefits of Electron's maturity, but you will have acrylic and mica *right now*, and a Chromium-based control that you can make interact with your app. It might work depending on your use case!

4. I cannot emphasize enough how early it seems for a lot of this "new" UI stuff. Even Edge Canary is still using an experimental flag to enable just mica (not even acrylic!) on Windows 11. As I said in my tldr; it'll likely be a while until these APIs stabilize and move from private to public... and trickle down to win32 apps through Project Reunion.

**Comment by TheColdPot** [comment]
1. Acrylic not just blur the background.
2. The performance of your App may be decreased by the backdrop-filter.
3. Chromium backdrop-filter can't access to your desktop
And so on.

**Comment by rmill777** [comment]
https://www.npmjs.com/package/glasstron

What about this then? Seems like it works, so clearly it is _possible_ 

However, if you try this package for yourself the acrylic setting seems buggy (on Windows 10 at least), use 'blurbehind' instead

**Comment by mlaurencin** [comment]
@alixsep Does the option `backgroundMaterial: 'acrylic'` option resolve this for you? More information can be found about the option in [this list](https://www.electronjs.org/docs/latest/api/browser-window#winsetbackgroundmaterialmaterial-windows) within the documentation.
